### PR TITLE
return 400 if the email address format is not valid

### PR DESCRIPTION
**Pull request by eishay**
@andrewconner @raymondng @2is10

### CODE REVIEW

**Comment by andrewconner** [comment]
Looks like `error` returns a `Forbidden`

**Comment by raymondng** [comment]
Just checking -- how do we handle this on the client side? Any changes required?

**Comment by eishay** [comment]
@andrewconner changed to BadRequest
@raymondng I don't think that they handle the different error checks now, though it will be good to do so

File: kifi-backend/modules/common/app/com/keepit/social/providers/UsernamePasswordProvider.scala [hunk 1]
```diff
@@ -29,19 +31,25 @@ class UsernamePasswordProvider(app: Application)
...
-            identity match {
-              case userIdentity: UserIdentity =>
-                log.info(s"[doAuth] userIdentity=$userIdentity")
-                if (passwordAuth.authenticate(userIdentity.userId.get, password)) Right(userIdentity) else Left(error("wrong_password"))
-              case _ =>
-                log.error(s"[doAuth] identity passed in is not of type <UserIdentity>; class=${identity.getClass}; obj=$identity")
-                Left(error("wrong_password")) // wrong_password for compatibility; auth_failure/internal_error more accurate
+        EmailAddress.validate(emailString.trim) match {
+          case Failure(e) =>
+            log.error(s"bad email format $emailString used for login", e)
```
Comment (2is10) [inline kifi-backend/modules/common/app/com/keepit/social/providers/UsernamePasswordProvider.scala]: A stack trace may not be particularly helpful in the log in this case.

**Comment by 2is10** [comment]
:+1:

**Pull request merged**